packages/components/Callout/macos/CalloutView.swift

```diff
@@ -175,14 +227,121 @@ class CalloutView: RCTView, CalloutWindowLifeCycleDelegate {
+				preconditionFailure("it should not be possible for the RCTTextShadowView to not be represented by an RCTTextView")
+		}
+
+		// Search for our target tag and get its startCharRange index in the overall Text view
+		let startCharRangeSearch = getStartCharRangeForTag(reactTag: subShadowView.reactTag, shadowView: textShadowView)
+		if (!startCharRangeSearch.found) {
+				// Did not find our reactTag
+				return nil
+		}
+
+		// Having found our target, return the bounding rect for its corresponding character range
+		return textView.getRectForCharRange(NSRange(location: startCharRangeSearch.startCharRange, length: getLengthOfTextShadowNode(shadowView: subShadowView)))
+	}
+
+	/// Get the leaf shadow view corresponding to a leaf Yoga node for the provided ShadowView
+	private func getLeafShadowViewForShadowView(shadowView: RCTShadowView?) -> RCTShadowView? {
+			var shadowParentIter = shadowView
+			while let shadowViewIter = shadowParentIter {
+					if (shadowViewIter.isYogaLeafNode()) {
+							return shadowViewIter
+					}
+					shadowParentIter = shadowViewIter.superview
+			}
+
+			return nil
+	}
+
+	// Return the anchor rect for the target prop if available
+	private func getAnchorRectForTarget() -> NSRect? {
+			guard let reactTag = target, let reactBridge = bridge else {
+				return nil
+		}
+
+		let zeroRect = CGRect(x: 0, y: 0, width: 0, height: 0)
+		let targetView = reactBridge.uiManager.view(forReactTag: reactTag)
+
+		// If the targetView is backed by an NSView and has a representative rect, return it as the anchor rect for the target
+		if let targetViewBounds = targetView?.bounds, !targetViewBounds.equalTo(zeroRect) {
+				return calculateAnchorViewScreenRect(anchorView: targetView, anchorBounds: targetViewBounds)
+		}
+
+		// If the targetView could not be found or was not a representative rect, it may be a child of a yoga leaf node e.g. virtualized text
+		guard let targetShadowView = reactBridge.uiManager.shadowView(forReactTag: target) else {
+				return nil
+		}
+
+		// Find the leaf ShadowView of our targetView
+		guard let leafShadowView = getLeafShadowViewForShadowView(shadowView: targetShadowView) else {
+			return nil
+		}
+
+		// Ensure we have a real NSView for our leaf ShadowView
+		guard let leafNSView = reactBridge.uiManager.view(forReactTag: leafShadowView.reactTag) else {
+			return nil
+		}
+
+		// Find the bounding rect of our targetView relative to the leafShadowView
+		guard let targetViewBounds = getBoundsForSubShadowOfLeafShadow(subShadowView: targetShadowView, leafShadowView: leafShadowView) else {
+			return nil
+		}
+
+		// If we could find the bounding rect of our target view and it's a representative rect, return it as the anchor rect for the target
+		if !targetViewBounds.equalTo(zeroRect) {
+				return calculateAnchorViewScreenRect(anchorView: leafNSView, anchorBounds: targetViewBounds)
+		}
+
+		// Unfortunately our efforts could not determine a valid anchor rect for our target prop
+		return nil
```

Comment:
Something feels bad about this.. but I'll have to think about it more and see if there's a more idiomatic way to do the same thing

Reply:
Can split to separate functions at a minimum

Reply:
I considered splitting the "TryGetBoundsAsNSBackedView" and "TryGetBoundsAsASubnodeOfALeafNode" portions into their own functions, it didn't make it a whole lot better.  I could be over-using guard conditions and stack a few `if let`s instead, but overall my dislikes seem mostly around the style and less in changing the general flow of the logic